// antd-mobile/lib/locale-provider/sv_SE.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _sv_Se = require('../date-picker-view/locale/sv_Se');

var _sv_Se2 = _interopRequireDefault(_sv_Se);

var _sv_Se3 = require('../date-picker/locale/sv_Se');

var _sv_Se4 = _interopRequireDefault(_sv_Se3);

var _sv_Se5 = require('../input-item/locale/sv_Se');

var _sv_Se6 = _interopRequireDefault(_sv_Se5);

var _sv_Se7 = require('../pagination/locale/sv_Se');

var _sv_Se8 = _interopRequireDefault(_sv_Se7);

var _sv_Se9 = require('../picker/locale/sv_Se');

var _sv_Se10 = _interopRequireDefault(_sv_Se9);

var _sv_Se11 = require('../search-bar/locale/sv_Se');

var _sv_Se12 = _interopRequireDefault(_sv_Se11);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

exports['default'] = {
    locale: 'sv',
    Pagination: _sv_Se8['default'],
    DatePicker: _sv_Se4['default'],
    DatePickerView: _sv_Se2['default'],
    InputItem: _sv_Se6['default'],
    Picker: _sv_Se10['default'],
    SearchBar: _sv_Se12['default']
};
module.exports = exports['default'];
